File: Backend-UOI/src/main/java/com/root/entity/OpinionInsight.java
package com.root.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "opinion_insights")
public class OpinionInsight {

    @Id
    private Integer id;

    @Column(name="product_name")
    private String productName;

    @Column(name="sentiment")
    private String sentiment;

    @Column(name="sentiment_confidence")
    private Double sentimentConfidence;

    @Column(name="sentiment_classified")
    private Integer sentimentClassified;

    @Column(name = "source")
    private String source;

    @Column(name="country")
    private String country;

    @Lob
    @Column(name = "content", columnDefinition = "MEDIUMTEXT")
    private String content;

    public String getContent() {
        return content;
    }

    public String getCountry() {
        return country;
    }

    public String getSource() {
        return source;
    }


    public Integer getId() {
        return id;
    }

    public String getProductName() {
        return productName;
    }

    public String getSentiment() {
        return sentiment;
    }

    public Double getSentimentConfidence() {
        return sentimentConfidence;
    }

    public Integer getSentimentClassified() {
        return sentimentClassified;
    }

}

File: genai-insights/app/models/opinion_insight.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    Text,
    Enum,
    Float,
    DateTime,
    DECIMAL,
    func,
    Date,
    ForeignKey
)
# from sqlalchemy.ext.declarative import declarative_base

# Base = declarative_base()

from app.db.base import Base

class OpinionInsight(Base):
    __tablename__ = "opinion_insights"

    # Primary Key
    id = Column(Integer, primary_key=True, index=True, autoincrement=True)

    # Timestamps
    creation_date = Column(
        DateTime,
        server_default=func.current_timestamp()
    )

    review_date = Column(
        Date,
        nullable=True
    )

    modification_timestamp = Column(
        DateTime,
        server_default=func.current_timestamp(),
        onupdate=func.current_timestamp()
    )

    # Core Review Fields (MANDATORY LOGICALLY)
    product_name = Column(String(40), nullable=True)
    source = Column(String(25), nullable=True)
    content = Column(Text, nullable=False)

    # Optional Metadata
    author = Column(String(255), nullable=True)
    country = Column(String(35), nullable=True)
    extra = Column(String(250), nullable=True)

    # Sentiment Fields
    sentiment = Column(
        Enum("positive", "negative", "neutral", name="sentiment"),
        nullable=True
    )

    sentiment_confidence = Column(
        DECIMAL(5, 4),
        nullable=True
    )

    rating = Column(Float, nullable=True)

    # Processing State Flags (CRITICAL)
    sentiment_classified = Column(Integer, nullable=False, default=0)
    component_classified = Column(Integer, nullable=False, default=0)

    # Component Context
    component = Column(String(25), nullable=True, default="overallphone")

    job_id = Column(
    Integer,
    ForeignKey(
        "job_status.id",
        ondelete="SET NULL",
        onupdate="CASCADE"
    ),
    nullable=True,
    index=True
    )
